### ISSUE TITLE
(🐞) False error "Expression type contains Any" from constructor when constructor isn't used

### ISSUE THREAD

**Issue by KotlinIsland** [python/mypy#13006]
```py
class A:
    def __init__(self, foo): ...

    @staticmethod
    def foo() -> None: ...

A.foo()  # error: Expression type contains "Any" (has type "Type[A]")
```
[playground](https://mypy-play.net/?mypy=latest&python=3.10&flags=disallow-any-expr&gist=cbc49b91397edc39f52a110c9d99eca6)

Despite the fact that `A` can be called as `(A, Any) -> Any`, the constructor isn't being referenced or invoked here.
